packages/encoding/src/rfc3339.ts

```diff
@@ -40,8 +40,13 @@ export function fromRfc3339(str: string): Date {
 
   const tzOffset = tzOffsetSign * (tzOffsetHours * 60 + tzOffsetMinutes) * 60; // seconds
 
-  const timestamp = Date.UTC(year, month - 1, day, hour, minute, second, milliSeconds) - tzOffset * 1000;
-  return new Date(timestamp);
+  const date = new Date();
+  date.setUTCFullYear(year);
+  date.setUTCMonth(month - 1);
+  date.setUTCDate(day);
```

Comment:
```suggestion
  date.setUTCFullYear(year, month - 1, day);
```

This should work too according to docs